util/compression.h

```diff
@@ -848,7 +848,7 @@ inline std::string ZSTD_TrainDictionary(const std::string& samples,
 #if ZSTD_VERSION_NUMBER >= 800  // v0.8.0+
   // skips potential partial sample at the end of "samples"
   size_t num_samples = samples.size() >> sample_len_shift;
-  std::vector<size_t> sample_lens(num_samples, 1 << sample_len_shift);
+  std::vector<size_t> sample_lens(num_samples, 1ULL << sample_len_shift);
```

Comment:
Can you use `size_t{1}` instead if it works?

Reply:
size_t is a variable size type so a better fix might be size_t(1)